tests/test_armv7cpu.py

```diff
@@ -1244,21 +1264,10 @@ def test_lsl_imm_carry(self):
         self.assertEqual(self.cpu.R2, 0x1 << 31)
         self._checkFlagsNZCV(1, 0, 1, 0)
 
-    def test_lslw(self):
-        ''' custom test for degenerate case'''
-        tmp_ks = Ks(KS_ARCH_ARM, KS_MODE_THUMB)
-        ords = tmp_ks.asm("lsl.w r5, r6, #3")[0]
-        asm = ''.join(map(chr, ords))
-        self.code = self.mem.mmap(0x1000, 0x1000, 'rwx')
-        self.data = self.mem.mmap(0xd000, 0x1000, 'rw')
-        self.stack = self.mem.mmap(0xf000, 0x1000, 'rw')
-        start = self.code + 4
-        self.mem.write(start, asm)
-        self.rf.write('PC', start)
-        self.rf.write('SP', self.stack + 0x1000)
-        self.rf.write('R5', 0x1)
-        self.rf.write('R6', 0x2)
-        self.cpu._set_mode(CS_MODE_THUMB)
+    @itest_setregs("R5=1", "R6=2")
+    @itest_custom_thumb("lsl.w r5, r6, #3")
```

Comment:
if you wanted, you could even make an `itest_thumb` as a counterpart to regular `itest`. the only difference is that the decorator calls cpu.execute so the test can simply assert.